src/impact-dashboard/FacilityInformation.tsx

```diff
@@ -80,46 +76,86 @@ export const AgeGroupGrid: React.FC<AgeGroupGridProps> = ({
   ...props
 }) => {
   const [collapsed, setCollapsed] = useState(collapsible);
+
+  const passedAgesKnown = (model: Record<string, any> | undefined) => {
+    if (model !== undefined) {
+      let keys = Object.keys(model);
+      return keys.some(function (key) {
+        return RegExp(/age\d+/).test(key);
+      });
+    } else {
+      return true;
+    }
+  };
+
+  const passedAgesUnknown = (model: Record<string, any> | undefined) => {
+    if (model !== undefined) {
+      let keys = Object.keys(model);
+      return keys.some(function (key) {
+        return RegExp(/ageUnknown\w+/).test(key);
+      });
+    } else {
+      return true;
+    }
+  };
+
+  const collapseAgeInputs = (model: Record<string, any> | undefined) => {
+    if (passedAgesKnown(model)) {
+      return false;
+    } else if (
+      passedAgesKnown(model) === false &&
+      passedAgesUnknown(model) === true
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  useEffect(() => {
+    setCollapsed(collapseAgeInputs(props.model));
+  }, []);
```

Comment:
@jessex and @macfarlandian I'm not sure if this is the best place for all this logic to live. Happy to move it around.

Reply:
seems right to me but I do think you want to add dependencies here so that the modal reacts to changing user data

Reply:
@macfarlandian adding `props.model` to the dependencies array was easy enough.

Adding that function call caused a rerender every time I clicked a toggle on show/hide age specific inputs. Does that work for you?

Reply:
oh yeah good point. obviously `props.model` is the important one here .... 

looking at the code I think you can probably move all of your new function definitions out of the render loop? It doesn't look like they actually have any dependencies on the local state or props of the component. Then you should be able to leave it out of the effect dependencies without triggering a lint warning and we'll be good to go

Reply:
@macfarlandian ok, I think it's ready for another look